models/User.go

```diff
@@ -949,6 +969,137 @@ func CancelOrder(userId uint32, orderId uint32, isAsk bool) (*Ask, *Bid, error)
 	}
 }
 
+// Helper function to determine what percentage the user should be taxed
+func getTaxPercent(netCash int64) uint64 {
+
+	keys := make([]int64, 0)
+	for taxBracket := range TaxBrackets {
+		keys = append(keys, taxBracket)
+	}
+	sort.Slice(keys, func(i, j int) bool { return keys[i] < keys[j] })
+
+	for _, taxBracket := range keys {
+		if netCash <= taxBracket {
+			return TaxBrackets[taxBracket]
+		}
+	}
+	return MaxTaxPercent
+}
+
+// Helper to calculate tax for bidding user
+func getTaxForBiddingUser(tx *gorm.DB, userId uint32, stockId uint32, l *logrus.Entry, stockQuantity uint64, stockPrice uint64, biddingUser *User) *TransactionSummary {
+	transactionSummary := &TransactionSummary{UserId: userId, StockId: stockId}
+	tx.First(&transactionSummary)
```

Comment:
(I think this should be fine, but still just check if you can make reads on a transaction object)

Reply:
What do you mean?

Reply:
Dont know why but I got confused about performing read queries on a tx object. Just wanted you to check if that works

Reply:
It works